src/CoreWCF.WebHttp/src/CoreWCF/Dispatcher/HttpStreamFormatter.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using CoreWCF.Channels;
+using CoreWCF.Description;
+
+namespace CoreWCF.Dispatcher
+{
+    internal class HttpStreamFormatter : IDispatchMessageFormatter, IClientMessageFormatter
+    {
+        readonly string _contractName;
+        readonly string _contractNs;
+        readonly string _operationName;
+
+        public HttpStreamFormatter(OperationDescription operation)
+        {
+            if (operation == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(operation));
+            }
+
+            _operationName = operation.Name;
+            _contractName = operation.DeclaringContract.Name;
+            _contractNs = operation.DeclaringContract.Namespace;
+        }
+
+        public object DeserializeReply(Message message, object[] parameters)
+        {
+            return GetStreamFromMessage(message, false);
+        }
+
+        public void DeserializeRequest(Message message, object[] parameters)
+        {
+            parameters[0] = GetStreamFromMessage(message, true);
+        }
+
+        public Message SerializeReply(MessageVersion messageVersion, object[] parameters, object result)
+        {
+            Message message = CreateMessageFromStream(result);
+            if (result == null)
+            {
+                SingleBodyParameterMessageFormatter.SuppressReplyEntityBody(message);
+            }
+
+            return message;
+        }
+
+        public Message SerializeRequest(MessageVersion messageVersion, object[] parameters)
+        {
+            Message message = CreateMessageFromStream(parameters[0]);
+            // TODO: Can I remove this?
+            //if (parameters[0] == null)
```

Comment:
I think this entire method goes away once you remove IClientMessageFormatter

Reply:
This was addressed in d6719a44244040967ed8059e2881340dae0cccac.